### PR TITLE
Multiple fixes for ModeSelect

### LINKED ISSUE #1402: ModeSelect6 fails with DEC Alpha system
# Info

- Which version of Pi are you using: 3
- Which github revision of software: HEAD (tag 23.11.01)
- Which board version: 2.3B
- Which computer is the PiSCSI connected to: DEC Personal Workstation 433au, running OpenVMS/Alpha 7.3
- Which OS you are using (output of 'lsb_release -a'): Debian 11 (bullseye)

# Describe the issue

Booting from piscsi fails in the SRM console (aka 'BIOS') with `failed to open`
Running with debug reveals
~~~
[2023-12-20 16:40:20.184] [debug] (ID:LUN 0:0) - Device is executing TestUnitReady ($00)
[2023-12-20 16:40:20.194] [debug] (ID:LUN 0:0) - Device is executing ModeSelect6 ($15)
[2023-12-20 16:40:20.194] [debug] (ID 0) - Error status: Sense Key $05, ASC $26
~~~

### PR DESCRIPTION
**Pull request by kkaempf**
This PR covers multiple issues

- The DEC Alpha SRM console (aka BIOS) calls ModeSelect6 with no payload

- OpenVMS Alpha calls ModeSelect6 with a page code of 1

- ModeSelect had an off-by-one error in computing the length

Fixes #1402

### CODE REVIEW

**Review by rdmark — CHANGES_REQUESTED** [review]
The code changes look reasonable to me, given your reading of the SCSI spec.

Can you please update the unit tests to get them passing with this new interpretation of the expected behavior? Please let us know if you need help getting set up with googletest locally.

**Comment by kkaempf** [comment]
@rdmark frankly, I'm at loss with fixing the tests :-( Is there any documentation ?

**Comment by rdmark** [comment]
@kkaempf The closest we have to documentation is the issue ticket where the tests were initially introduced! https://github.com/PiSCSI/piscsi/issues/794

Running the tests should be as easy as installing the googletest libraries, and then running `make test`. You can also see the failing CI jobs here in this PR.

The actual test code should be fairly self-explanatory I think, but let me know if you need further pointers ;)

**Comment by kkaempf** [comment]
The test output is quite unhelpful 😞 
```
[ RUN      ] ScsiCommandUtilTest.ModeSelect6
unknown file: Failure
C++ exception with description "std::exception" thrown in the test body.
[  FAILED  ] ScsiCommandUtilTest.ModeSelect6 (0 ms)

```

**Comment by kkaempf** [comment]
Tests fixed (somewhat hackish imho)

File: cpp/test/scsi_command_util_test.cpp [hunk 1]
```diff
@@ -62,7 +67,15 @@ TEST(ScsiCommandUtilTest, ModeSelect6)
 			Property(&scsi_exception::get_asc, asc::invalid_field_in_parameter_list))))
 		<< "Not enough command parameters";
 
-	EXPECT_FALSE(ModeSelect(scsi_command::eCmdModeSelect6, cdb, buf, LENGTH, 512).empty());
+	// check length computation
+	buf[3] = 8;
+	buf[10] = 2;
+	buf[12] = 1;
+	buf[13] = 10,
```
Comment (rdmark) [inline cpp/test/scsi_command_util_test.cpp:74]: Should end with a semicolon, not a comma. :)

**Comment by rdmark** [comment]
@kkaempf What makes you say the tests are hackish?

**Comment by kkaempf** [comment]
> @kkaempf What makes you say the tests are hackish?

I can't say (yet ?) that I actually know what I'm doing there 🙈

**Comment by kkaempf** [comment]
This should be complete now I believe.

**Review by rdmark — APPROVED** [review]
Thanks for your contribution!

**Comment by rdmark** [comment]
Rebasing with the `develop` branch. This is where all new code goes. We then rebase with `main` after each stable release.

**Pull request merged**